packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,394 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_tools/src/base/process.dart';
+import 'package:flutter_tools/src/base/terminal.dart';
+import 'package:meta/meta.dart';
+import 'package:vm_snapshot_analysis/treemap.dart';
+
+import '../base/file_system.dart';
+import '../convert.dart';
+import 'logger.dart';
+
+class SizeAnalyzer {
+  SizeAnalyzer(this._fileSystem, this._logger, this._processUtils);
+
+  final FileSystem _fileSystem;
+  final Logger _logger;
+  final ProcessUtils _processUtils;
+
+  static const String aotSizeFileName = 'aot-size.json';
+
+  static const int tableWidth = 80;
+
+  Future<Map<String, dynamic>> analyzeApkSize({
+    @required File apk,
+    @required File aotSizeJson,
+  }) async {
+    _logger.printStatus('▒' * tableWidth);
+    _printEntitySize(
+      '${apk.basename} (total compressed)',
+      apk.lengthSync(),
+      0,
+      showColor: false,
+    );
+    _logger.printStatus('━' * tableWidth);
+    final Directory tempApkContent = _fileSystem.systemTempDirectory.createTempSync('flutter_tools.');
+    // TODO: implement Windows.
+    String unzipOut;
+    try {
+      unzipOut = (await _processUtils.run(<String>[
+        'unzip',
+        '-o',
+        '-v',
+        apk.path,
+        '-d',
+        tempApkContent.path
+      ])).stdout;
+    } on Exception catch (e) {
+      print(e);
+    } finally {
+      // We just want the the stdout printout. We don't need the files.
+      tempApkContent.deleteSync(recursive: true);
+    }
+
+    final SymbolNode apkAnalysisRoot = _parseUnzipFile(unzipOut);
+
+    for (final SymbolNode firstLevelPath in apkAnalysisRoot.children) {
+      _printEntitySize(
+        firstLevelPath.name,
+        firstLevelPath.value,
+        1,
+      );
+      if (firstLevelPath.name == 'lib') {
+        _printLibDetails(firstLevelPath, '', aotSizeJson);
+      }
+    }
+
+    _logger.printStatus('▒' * tableWidth);
+    
+    Map<String, dynamic> apkAnalysisJson = apkAnalysisRoot.toJson();
+    
+    // TODO(peterdjlee): Add aot size for all platforms.
+    apkAnalysisJson = addAotSizeDataToJson(
+      apkAnalysisJson,
+      'lib/arm64-v8a/libapp.so (Dart AOT)'.split('/'), 
```

Comment:
add code comments here for what this is doing

Reply:
Done